### PR TITLE
[bugfix] remove `ordinal` for `Turkish` locale as they use `cardin…

### PR DESCRIPTION
**Pull request by alan-agius4**
Turkish don't use ordinal days but rather cardinals

https://github.com/moment/moment/issues/4122#issuecomment-353021460

### CODE REVIEW

**Comment by marwahaha** [comment]
@BYK  - can you review this? 

It's unclear if we should keep the Turkish ordinals in the codebase at all (this PR removes them entirely).

File: src/test/locale/tr.js [hunk 1]
```diff
@@ -22,20 +22,20 @@ test('parse', function (assert) {
...
+            ['d do dddd ddd dd',                   '0 0 Pazar Paz Pz'],
+            ['DDD DDDo DDDD',                      '45 45 045'],
+            ['w wo ww',                            '7 7 07'],
             ['h hh',                               '3 03'],
             ['H HH',                               '15 15'],
             ['m mm',                               '25 25'],
             ['s ss',                               '50 50'],
             ['a A',                                'pm PM'],
-            ['[yılın] DDDo [günü]',                'yılın 45\'inci günü'],
+            ['[yılın] DDDo [günü]',                'yılın 45 günü'],
```
Comment (BYK) [inline src/test/locale/tr.js]: This is incorrect. The old test is correct and we should keep that.

Comment (alan-agius4) [reply src/test/locale/tr.js]: is this the only spec that should be kept?

Comment (BYK) [reply src/test/locale/tr.js]: As mentioned in the original issue earlier, one never uses "December 3rd", it is always "3 December" or rarely, "December 3". Ordinals are only used when you are saying something like "The 3rd week of the year", "The 5th day of the month" (this one actually uses a slightly different thing than ordinals but ordinals are also fine) or "the first Sunday of March".

Comment (alan-agius4) [reply src/test/locale/tr.js]: Did some further changes, can you re-check please? I am not sure if I covered everything.

**Comment by marwahaha** [comment]
@alan-agius4  - could you add a couple tests specific to the issue you're fixing?

@BYK - could you review again?

**Comment by alan-agius4** [comment]
I actually amended the existing test to match the new behaviour for some dates

**Pull request merged**